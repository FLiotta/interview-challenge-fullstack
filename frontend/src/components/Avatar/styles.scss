@import '../../styles/colors.scss';

.avatar {
  border-radius: 50%;
  border: 2px solid $brand_color;
  padding: 2px;
  display: block;

  &__fill {
    width: 100%;
    height: 100%;
    border-radius: 50%;

    &--0 {
      background: #f12711;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to bottom, #f5af19, #f12711);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to bottom, #f5af19, #f12711); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--1 {
      background: #3D7EAA;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to bottom, #FFE47A, #3D7EAA);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to bottom, #FFE47A, #3D7EAA); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--2 {
      background: #ff9966;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #ff5e62, #ff9966);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #ff5e62, #ff9966); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--3 {
      background: #43C6AC;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #F8FFAE, #43C6AC);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #F8FFAE, #43C6AC); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--4 {
      background: #EECDA3;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #EF629F, #EECDA3);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #EF629F, #EECDA3); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--5 {
      background: #556270;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #FF6B6B, #556270);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #FF6B6B, #556270); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--6 {
      background: #e43a15;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #e65245, #e43a15);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #e65245, #e43a15); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--7 {
      background: #ffe259;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #ffa751, #ffe259);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #ffa751, #ffe259); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }

    &--8 {
      background: #9796f0;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #fbc7d4, #9796f0);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #fbc7d4, #9796f0); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */ 
    }

    &--9 {
      background: #ec008c;  /* fallback for old browsers */
      background: -webkit-linear-gradient(to left, #fc6767, #ec008c);  /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(to left, #fc6767, #ec008c); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */      
    }
  }
}