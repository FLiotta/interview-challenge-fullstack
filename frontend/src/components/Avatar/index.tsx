// @Project
import React from 'react';

// @Own
import './styles.scss';

interface IProps {
  userId?: number,
  width?: number | string,
  height?: number | string
}

const Avatar: React.FC<IProps> = ({
  userId = 1,
  width,
  height
}) => {
  return (
    <div
      data-testid="avatar-testid"
      className="avatar" 
      style={{ width, height }}
    >
      <div
        className={`avatar__fill avatar__fill--${userId}`}
        data-testid="avatar-fill-testid"
      >
      </div> 
    </div>
  )
}

export default Avatar;