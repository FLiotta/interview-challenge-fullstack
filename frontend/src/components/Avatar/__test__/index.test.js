// @Package
import { render, getByTestId } from '@testing-library/react';

// @Project
import Avatar from '../index';

describe('Avatar Component', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Avatar />);

    expect(container).toMatchSnapshot();
  })

  it('Receives "width" prop', () => {
    const WIDTH = 50;
    
    const { container } = render(<Avatar width={WIDTH} />);

    const avatar = getByTestId(container, 'avatar-testid')

    expect(avatar.style.width).toBe(`${WIDTH}px`)
  })

  it('Receives "height" prop', () => {
    const HEIGHT = 50;
    
    const { container } = render(<Avatar height={HEIGHT} />);

    const avatar = getByTestId(container, 'avatar-testid')

    expect(avatar.style.height).toBe(`${HEIGHT}px`)
  })

  it('Receives "userId" prop', () => {
    const USERID = 500;

    const { container } = render(<Avatar userId={USERID} />)

    const avatarFill = getByTestId(container, 'avatar-fill-testid');

    expect(avatarFill).toHaveClass(`avatar__fill--${USERID}`);
  })
})