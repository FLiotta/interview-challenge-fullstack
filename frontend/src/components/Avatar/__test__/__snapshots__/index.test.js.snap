// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Should match snapshot 1`] = `
<div>
  <div
    class="avatar"
    data-testid="avatar-testid"
  >
    <div
      class="avatar__fill avatar__fill--1"
      data-testid="avatar-fill-testid"
    />
  </div>
</div>
`;
